enhancement for postgresql backup instance

ms-henglu opened the pull request:
![image](https://user-images.githubusercontent.com/79895375/131796275-7411d4bf-02da-42c9-a660-e8b33ad95a1f.png)

stephybun commented on website/docs/r/data_protection_backup_instance_postgresql.html.markdown:
```diff
@@ -86,6 +151,8 @@ The following arguments are supported:
 
 * `backup_policy_id` - (Required) The ID of the Backup Policy.
 
+* `database_credential_key_vault_secret_id` - (Optional) The ID of the key vault secret where stores the connection string of the database.
```
```suggestion
* `database_credential_key_vault_secret_id` - (Optional) The ID of the key vault secret which stores the connection string of the database.
```

stephybun reviewed (changes requested):
Thanks @ms-henglu. Aside from a few minor comments this is looking good 👍🏼. Since this is my first larger PR review I may ask someone from my team have a look over this as well.

ms-henglu commented on internal/services/dataprotection/data_protection_backup_instance_postgresql_resource.go:
```diff
@@ -184,6 +202,13 @@ func resourceDataProtectionBackupInstancePostgreSQLRead(d *schema.ResourceData,
 			d.Set("backup_policy_id", props.PolicyInfo.PolicyID)
 		}
+		if props.DatasourceAuthCredentials != nil {
+			if credential, ok := props.DatasourceAuthCredentials.AsSecretStoreBasedAuthCredentials(); ok {
```
I think it's better not to return an error. It may be other kinds of credentials which are not supported yet.

manicminer replied:
If not returning an error, maybe we should at least log a debug message since this is otherwise a silent failure.

ms-henglu commented:
@stephybun , Hi, thanks for code review!

I've updated it according to your suggestions, please take another look, thanks!

manicminer commented on internal/services/dataprotection/data_protection_backup_instance_postgresql_resource.go:
```diff
@@ -68,6 +69,12 @@ func resourceDataProtectionBackupInstancePostgreSQL() *pluginsdk.Resource {
 				ValidateFunc: validate.BackupPolicyID,
 			},
+
+			"database_credential_key_vault_secret_id": {
```
This could be suffixed to match the attribute from the key_vault_secret resource

```suggestion
			"database_credential_key_vault_secret_versionless_id": {
```

manicminer commented on website/docs/r/data_protection_backup_instance_postgresql.html.markdown:
```diff
@@ -86,6 +151,8 @@ The following arguments are supported:
 
 * `backup_policy_id` - (Required) The ID of the Backup Policy.
 
+* `database_credential_key_vault_secret_id` - (Optional) The ID of the key vault secret which stores the connection string of the database.
```
(for clarity)

```suggestion
* `database_credential_key_vault_secret_versionless_id` - (Optional) The versionless ID of the key vault secret which stores the connection string of the database.
```

manicminer reviewed (changes requested):
Hi @ms-henglu, agreed with @stephybun - pending from some minor comments and test results this should be good to merge 👍

ms-henglu commented:
@manicminer , Hi, thanks for code review.

I've added debug message when conversion failed. And the field supports both key vault secret's id and versionless id, so I've added that to the doc. Thanks!

ms-henglu commented:
About the test results, service team is fixing the bug which caused all tests failed. I'll attach a tc results once it got fixed.

ms-henglu commented:
@manicminer , Hi, tests passed. Could you please take another look, thanks!

manicminer reviewed (approved):
Thanks @ms-henglu, this LGTM 🔌 

Pasting test results:
![Screenshot 2021-09-02 at 10 06 08](https://user-images.githubusercontent.com/251987/131816265-ff16ab39-6c1a-4285-9632-a48d5b312b5d.png)
